PhoneBook.Web/Classes/EditFormPage.cs

```diff
@@ -85,47 +74,47 @@ protected bool SaveObject()
       EnsurePostLoadInvoked();
       EnsureValidatableControlsInitialized();
 
-      bool isValid = DataSource.Validate();
-      Control firstInvalidControl = null;
+      var isValid = DataSource.Validate();
 
-      foreach (DataEditUserControl control in DataEditUserControls)
+      var firstInvalidControl = DataEditUserControls.FirstOrDefault(control => !control.Validate());
+      if (firstInvalidControl != null)
       {
-        if (! control.Validate())
-        {
-          isValid = false;
-          firstInvalidControl = control;
-          break;
-        }
+        isValid = false;
       }
-
+      
       if (isValid)
       {
-        foreach (DataEditUserControl control in DataEditUserControls)
-          control.SaveValues (false);
-
-        DataSource.SaveValues (false);
-        _isSaved = true;
-        // transaction will be committed by caller or using auto commit.
+        SaveValidObject();
       }
       else
       {
-        TabbedMultiView multiView = UserControlMultiView;
-        if (multiView != null)
+        var multiView = UserControlMultiView;
+        if (multiView == null)
+          return false;
+
+        for (Control control = firstInvalidControl; control != null; control = control.Parent)
         {
-          for (Control control = firstInvalidControl; control != null; control = control.Parent)
-          {
-            if (control is TabView)
-            {
-              multiView.SetActiveView ((TabView) control);
-              break;
-            }
-          }
+          if (!(control is TabView))
```

Comment:
The preference is to write positive statements (unless it's an exit condition, which this can argued here) and to avoid continue unless there's a compelling reason. With the new code, we have two different loop-control mechanism, which should definitely be avoided for easier reasoning about the code.